nabla-indexer/test/indexer_tests.ts

```diff
@@ -0,0 +1,1067 @@
+        {
+          reserve: "11000000000000",
+          reserveWithSlippage: "11000000000000",
+          totalLiabilities: "11000000000000",
+          totalSupply: "11000000000000",
+          tokenName: "Token 1",
+        },
+        {
+          reserve: "12000000000000",
+          reserveWithSlippage: "12000000000000",
+          totalLiabilities: "12000000000000",
+          totalSupply: "12000000000000",
+          tokenName: "Token 2",
+        },
+        {
+          reserve: "0",
+          reserveWithSlippage: "0",
+          totalLiabilities: "0",
+          totalSupply: "0",
+          tokenName: "Token Extra 2",
+        },
+      ]);
+    },
+
+    async testReplaceSwapPool() {
+      const instance = await createNablaInstance(unit(10), [unit(11), unit(12)]);
+      let indexerRouters = (await readIndexerUntil(
+        (routers) => routers.find((router) => router.id === address(instance.router)) !== undefined,
+        3000
+      ))!;
+      assertTrue(indexerRouters !== undefined);
+
+      verifyIndexer(indexerRouters, instance, [
+        {
+          reserve: "11000000000000",
+          reserveWithSlippage: "11000000000000",
+          totalLiabilities: "11000000000000",
+          totalSupply: "11000000000000",
+          tokenName: "Token 1",
+        },
+        {
+          reserve: "12000000000000",
+          reserveWithSlippage: "12000000000000",
+          totalLiabilities: "12000000000000",
+          totalSupply: "12000000000000",
+          tokenName: "Token 2",
+        },
+      ]);
+
+      const replacementPool = await newSwapPool(
+        address(instance.assets[1]),
+        address(instance.nablaCurve),
+        address(instance.router),
+        address(instance.backstop),
+        0,
+        `LP New 1`,
+        `LPN1`
+      );
+
+      await instance.assets[1].approve(address(replacementPool), MAX_UINT256);
+      await replacementPool.deposit(unit(20));
+
+      await new Promise((resolve) => setTimeout(resolve, 3000));
+      indexerRouters = await readIndexer();
+
+      verifyIndexer(indexerRouters, instance, [
+        {
+          reserve: "11000000000000",
+          reserveWithSlippage: "11000000000000",
+          totalLiabilities: "11000000000000",
+          totalSupply: "11000000000000",
+          tokenName: "Token 1",
+        },
+        {
+          reserve: "12000000000000",
+          reserveWithSlippage: "12000000000000",
+          totalLiabilities: "12000000000000",
+          totalSupply: "12000000000000",
+          tokenName: "Token 2",
+        },
+      ]);
+
+      await instance.router.registerPool(address(instance.assets[1]), address(replacementPool));
+
+      instance.swapPools[0] = replacementPool;
+      indexerRouters = (await readIndexerUntil(
+        (routers) =>
+          routers
+            .find((router) => router.id === address(instance.router))
+            ?.swapPools.find((swapPool) => swapPool.id === address(replacementPool)) !== undefined,
```

Comment:
Should we also check that the replaced `swapPool` does not exist anymore (not registered with the router)?

Reply:
Yes, I think this is happening here already the `verifyIndexer` that comes up next checks whether there are only two swap pools. Since the first swap pool's reserve changed from `11000000000000` to `20000000000000` and there are only two pools, then this would mean that the indexer does not report the replaced swap pool anymore.